### PR TITLE
Workaround: get logs on a best-effort basis (handle epoch-change edge-case)

### PR DESCRIPTION
**Pull request by andreibancioiu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- In some rare cases, right at the beginning of an epoch, `logs & events` of a transaction can we mistakenly saved in the storage associated with the previous epoch.
  
## Proposed Changes
- In this PR, we attempt to load `logs & events` both from the requested epoch `N`, and from the epoch `N-1`.

## Testing procedure
- Regular testing
- `rosetta-cli` checker (with ESDT) against devnet (e.g. shard 0)

### CODE REVIEW

**Review by iulianpascalau — APPROVED** [review]

File: node/external/logs/logsRepository.go [hunk 1]
```diff
@@ -24,7 +24,23 @@ func newLogsRepository(storageService dataRetriever.StorageService, marshaller m
 	}
 }
 
+// getLog loads event & logs on a best-effort basis.
+// It searches for logs both in epoch N and epoch N-1, in order to overcome epoch-changing edge-cases.
 func (repository *logsRepository) getLog(logKey []byte, epoch uint32) (*transaction.Log, error) {
+	txLog, err := repository.doGetLog(logKey, epoch)
+	if err != nil {
+		txLog, errPreviousEpoch := repository.doGetLog(logKey, epoch-1)
```
Comment (bogdan-rosianu) [inline node/external/logs/logsRepository.go:32]: might check that epoch is not 0 before subtracting 1 from it

Comment (andreibancioiu) [reply node/external/logs/logsRepository.go:32]: Good catch :rocket: Fixed.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

File: node/external/logs/logsRepository.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ func newLogsRepository(storageService dataRetriever.StorageService, marshaller m
 // It searches for logs both in epoch N and epoch N-1, in order to overcome epoch-changing edge-cases.
 func (repository *logsRepository) getLog(logKey []byte, epoch uint32) (*transaction.Log, error) {
 	txLog, err := repository.doGetLog(logKey, epoch)
-	if err != nil {
+	if err != nil && epoch > 0 {
```
Comment (iulianpascalau) [inline node/external/logs/logsRepository.go:31]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**